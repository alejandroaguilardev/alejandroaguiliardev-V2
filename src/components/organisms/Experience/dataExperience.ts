export const dataExperience = [
	{
		title: "QOLKREX FOUNDATION",
		subtitle: "BLOCKCHAIN DEVELOPER",
		description:
			"Foundation that promotes Blockchain research and development in Peru.",
		icon: "/img/experiences/qolkrex.png",
	},
	{
		title: "MEDIWEB",
		subtitle: "DEVELOPER FULL STACK",
		description:
			"Company specialized in solutions for clinics in order to provide comprehensive management of the operations they perform.",
		icon: "/img/experiences/mediweb.png",
	},
	{
		title: "GWEB7",
		subtitle: "DEVELOPER FRONT END",
		description:
			"Marketing agency that offers multiple quality services with creativity.",
		icon: "/img/experiences/gweb7.png",
	},
];
