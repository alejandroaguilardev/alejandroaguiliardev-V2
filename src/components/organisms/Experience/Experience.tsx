import { Grid, Container, Box } from "@mui/material"
import { dataExperience } from "./dataExperience"
import { TitleSection } from "../../atoms"
import { CardItem } from "../CardItem/CardItem"

export const Experience = () => {
    return (
        <Container maxWidth="lg">
            <TitleSection
                title="My Work Experience"
                description={["Featured companies which I have worked in my career as a developer."]}
            />

            <Grid container spacing={5} mt={2}>
                {dataExperience.map((data) => (
                    <Grid item xs={12} md={4} key={data.title}>
                        <CardItem
                            title={data.title}
                            subtitle={data.subtitle}
                            description={data.description}
                            icon={data.icon}
                            size='medium'
                        />
                    </Grid>
                ))}
            </Grid>
        </Container >
    )
}
