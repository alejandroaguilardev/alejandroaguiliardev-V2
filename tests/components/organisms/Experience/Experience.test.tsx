import { render, screen } from "@testing-library/react";
import { Experience } from "../../../../src/components/organisms/Experience/Experience";

describe("Experience", () => {
	test("toMatchSnapshot", () => {
		const { container } = render(<Experience />);

		expect(container).toMatchSnapshot();
        expect(screen.getByText("My Work Experience")).toBeTruthy();
        expect(screen.getByText("DEVELOPER FRONT END")).toBeTruthy();
	});
});
