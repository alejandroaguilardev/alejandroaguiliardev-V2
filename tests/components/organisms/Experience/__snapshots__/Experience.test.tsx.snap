// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience toMatchSnapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-1yjvs5a"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-162zxzj-MuiTypography-root"
        role="title"
      >
        My Work Experience
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1mqboxc-MuiTypography-root"
        role="description"
      >
        Featured companies which I have worked in my career as a developer.
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 css-6kn84p-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1uvn853-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiBox-root css-18i215d"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27100%27%20height=%27100%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="QOLKREX FOUNDATION"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="MuiBox-root css-gajhq5"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-yg6atj-MuiTypography-root"
            >
              QOLKREX FOUNDATION
            </h3>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1a3kej9-MuiTypography-root"
            >
              BLOCKCHAIN DEVELOPER
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-r5yzcc-MuiTypography-root"
            >
              Foundation that promotes Blockchain research and development in Peru.
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1uvn853-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiBox-root css-18i215d"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27100%27%20height=%27100%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="MEDIWEB"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="MuiBox-root css-gajhq5"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-yg6atj-MuiTypography-root"
            >
              MEDIWEB
            </h3>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1a3kej9-MuiTypography-root"
            >
              DEVELOPER FULL STACK
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-r5yzcc-MuiTypography-root"
            >
              Company specialized in solutions for clinics in order to provide comprehensive management of the operations they perform.
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1uvn853-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiBox-root css-18i215d"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27100%27%20height=%27100%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="GWEB7"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="MuiBox-root css-gajhq5"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-yg6atj-MuiTypography-root"
            >
              GWEB7
            </h3>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1a3kej9-MuiTypography-root"
            >
              DEVELOPER FRONT END
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-r5yzcc-MuiTypography-root"
            >
              Marketing agency that offers multiple quality services with creativity.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
